### PR TITLE
[Enhancement] Add tablet_sched_colocate_balance_after_system_stable_time_s to avoid unnecessary colocate balance

### PR DESCRIPTION
**Pull request by gengjun-git**
## Why I'm doing:
Colocate balance is a very time-consuming operation,
and our system should try to avoid triggering colocate balance.
In the BYOC environment, customers will stop all machines when the cluster is not in use to save machine resources.
When the machine is started again, unnecessary colocate balance
will be triggered due to the inconsistent start time of the machines.
To avoid this situation, we introduced the tablet_sched_colocate_balance_after_system_stable_time_s parameter.
If the status(alive and decommissioned) of all backend can maintain consistency within
tablet_sched_colocate_balance_after_system_stable_time_s, then the colocate balance will be triggered.
Default value is 15min.

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by murphyatwork — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/clone/ColocateTableBalancer.java [hunk 1]
```diff
@@ -246,13 +249,34 @@ public String toString() {
...
+        if (!Config.tablet_sched_disable_colocate_balance && isSystemStable(GlobalStateMgr
+                .getCurrentState().getNodeMgr().getClusterInfo())) {
             relocateAndBalancePerGroup();
             relocateAndBalanceAllGroups();
         }
         matchGroups();
     }
 
+    /**
+     * If the availableBackendIds can maintain consistency within
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/clone/ColocateTableBalancer.java:261]: unchanged

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 2]
```diff
@@ -1423,6 +1423,20 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true, aliases = {"disable_colocate_balance"})
     public static boolean tablet_sched_disable_colocate_balance = false;
 
+    /**
+     * Colocate balance is a very time-consuming operation,
+     * and our system should try to avoid triggering colocate balance.
+     * In the BYOC environment, customers will stop all machines when the cluster is not in use to save machine resources.
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1429]:
```suggestion
     * In some situation, customers will stop all machines when the cluster is not in use to save machine resources.
```

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 3]
```diff
@@ -1423,6 +1423,20 @@ public class Config extends ConfigBase {
...
+     * When the machine is started again, unnecessary colocate balance
+     * will be triggered due to the inconsistent start time of the machines.
+     * To avoid this situation, we introduced the tablet_sched_colocate_balance_after_system_stable_time_s parameter.
+     * If the status(alive and decommissioned) of all backend can maintain consistency within
+     * tablet_sched_colocate_balance_after_system_stable_time_s, then the colocate balance will be triggered.
+     * Default value is 15min.
+     */
+    @ConfField(mutable = true)
+    public static long tablet_sched_colocate_balance_after_system_stable_time_s = 15 * 60;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1438]:
```suggestion
    public static long tablet_sched_colocate_balance_wait_system_stable_time_s = 15 * 60;
```

File: fe/fe-core/src/main/java/com/starrocks/clone/ColocateTableBalancer.java [hunk 4]
```diff
@@ -246,13 +249,34 @@ public String toString() {
...
+     * tablet_sched_colocate_balance_after_system_stable_time_s, the system is considered stable.
+     */
+    protected boolean isSystemStable(SystemInfoService infoService) {
+        Set<Long> currentAvailableBackendIds = new HashSet<>(infoService.getAvailableBackendIds());
+        if (!currentAvailableBackendIds.equals(availableBackendIds)) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/clone/ColocateTableBalancer.java:266]: should only affect first restart?

**Review by nshangyiming — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**